Update impersonation docs

zmb3 opened the pull request:
Remove duplicate titles, fix rendering of unordered lists, and take a pass at better explaining what some of the more advanced impersonation rules are doing.

Fixes #7988

zmb3 committed:
Update impersonation docs

Remove duplicate titles, fix rendering of unordered lists, and take a
pass at better explaining what some of the more advanced impersonation
rules are doing.

Fixes #7988

zmb3 commented on docs/pages/access-controls/guides/impersonation.mdx:
```diff
@@ -133,26 +142,33 @@ $ ssh -J jenkins@teleport.localhost:3023 -p 3022 jenkins@127.0.0.1
-- Alice can get a new `jenkins` certificate with the same TTL, but metadata updated, for example
-pointing to a different Teleport leaf cluster.
-Teleport allows impersonated users to renew their certificates with the reduced scope of the certificate.
+- Even though Alice's `max_session_ttl` is 10 hours, she can issue a cert with a
```
@klizhentas could use your review here. I don't fully understand what this means. Allowing Alice to issue a cert with a *longer* TTL does feel like it is _limiting_ the TTL, it's actually extending it. Is this correct?

klizhentas replied:
yes, better would say that certificate's TTL is extended to the `max_session_ttl` of the roles that are being impersonated.

zmb3 replied:
Ok understood. I will update this section.

zmb3 committed:
Fold the long 'where' rules.

We want the YAML parser to ignore the newline, the line is only
wrapped for readability.

klizhentas commented on docs/pages/access-controls/guides/impersonation.mdx:
```diff
@@ -19,12 +20,15 @@ CI/CD user Jenkins and a security scanner.
 (!docs/pages/includes/tctl.mdx!)
 
-## Step 1/3. Create a CI/CD user
+## Step 1: Create a CI/CD user and corresponding role
```
The `1/n` process is to show user the overall amount of steps (small) required to complete the section.

zmb3 replied:
Yeah I see that,I removed it thinking it would be easy to get out of date if steps are added or removed. (As evidenced by the fact that we had multiple "Step 3/3" before this PR)

The table of contents on the right hand side of the doc page will still show 3 steps, so I think that shows that it's a small number of steps, but happy to revert if that's what you prefer. Let me know.

klizhentas commented on docs/pages/access-controls/guides/impersonation.mdx:
```diff
@@ -94,14 +101,16 @@ spec:
-and impersonate `jenkins`. We assigned a preset `access` role that allows users to access clusters for
-Alice's convenience.
+Next, create an interactive user `alice` and assign the impersonator role so
+that `alice` can impersonate `jenkins`.who can Note: we also assign the preset
```
Let's use Admonition type=note for notes

klizhentas commented on docs/pages/access-controls/guides/impersonation.mdx:
```diff
@@ -5,12 +5,13 @@ description: How to issue short-lived certs on behalf of Teleport users using im
-Sometimes users need to create short-lived certificates for non-interactive users, for example, CI/CD systems.
-Your programs interacting with Teleport may need to create their own authentication as well.
-Teleport's impersonation allows users and robots to create short-lived certs for other users and roles.
+Sometimes users need to create short-lived certificates for non-interactive
```
Thanks for the PR! As you started contributing to the docs, check out our `meta` guide on style here:

https://goteleport.com/docs/docs/

klizhentas commented on docs/pages/access-controls/guides/impersonation.mdx:
```diff
@@ -244,8 +271,8 @@ spec:
 
-This role will work with SSO users, but we can test locally, by updating Alice's account
-entry with traits:
+While user traits typically from from an external identity provider, we can test
```
`from from` -> `from`

klizhentas commented on docs/pages/access-controls/guides/impersonation.mdx:
```diff
@@ -244,8 +271,8 @@ spec:
-This role will work with SSO users, but we can test locally, by updating Alice's account
-entry with traits:
+While user traits typically from from an external identity provider, we can test
+with local user `alice` by manually updating Alices's account with traits.
```
with a local user?

zmb3 replied:
I'm not sure what this comment means. The guide has us manually updating spec.traits for a local user, right?

klizhentas replied:
Sorry, I was wondering if `a` article is omitted in this sentence: "we can test with **a** local user, ...

klizhentas reviewed (changes requested):
Thanks for the PR, check out some comments.

zmb3 committed:
Address feedback

- use adminition for notes
- fix typo
- clarify the max TTL for impersonated roles

zmb3 committed:
Re-add step numbers

klizhentas commented:
@zmb3 LGTM, can you please also send a backport PR to `branch/v7`

zmb3 commented:
> @zmb3 LGTM, can you please also send a backport PR to `branch/v7`

Yep, will do!

zmb3 committed:
Merge branch 'master' into zmb3/impersonation-docs
